### PR TITLE
[Home][ArtworkRail] Support for landscape iPad layout

### LINKED ISSUE #280: [Home][ArtworksRail] Landscape iPad layout should have 4 columns
See https://github.com/artsy/eigen/issues/1441

### PR DESCRIPTION
**Pull request by sarahscott**
There's a small delay for image-loading but it could be because it's in the simulator:

rotate

- [x] Get 4 columns/3 columns working
- [x] Make transition smoother
- [x] Edge cases: does it look nice pre-expansion? post-expansion?
- [x] Margins

Fixes #280

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="true">No CHANGELOG added.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">The following files do not have tests: <a href="https://github.com/artsy/emission/blob/8a00e472c77d8ae5388bab69776aa740aa874a60/lib/components/artwork_grids/generic_grid.js">lib/components/artwork_grids/generic_grid.js</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: lib/components/artwork_grids/generic_grid.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ import Artwork from './artwork'
 class GenericArtworksGrid extends React.Component {
   state: {
     sectionDimension: number,
+    sectionCount: number,
   };
 
   static defaultProps = {
```
Comment (alloy) [inline lib/components/artwork_grids/generic_grid.js]: I guess we should make `sectionCount` in the props configurable for portrait and landscape for it to even make sense to be a prop. For now, though, maybe we should just remove the prop, as I don’t believe it’s used anywhere anyways.

Comment (sarahscott) [inline lib/components/artwork_grids/generic_grid.js]: aye

File: lib/components/artwork_grids/generic_grid.js [hunk 2]
```diff
@@ -117,7 +130,6 @@ const styles = StyleSheet.create({
     flexDirection: 'row',
   },
   section: {
-    flex: 1,
```
Comment (sarahscott) [inline lib/components/artwork_grids/generic_grid.js]: This was making the sections expand to fill the screen, making for a really weird transition. Eventually the layout would settle into 3 or 4 columns but with this removed, it looks much better.

**Comment by sarahscott** [comment]
Follow button needs some 💙  but I'll do that in its own PR.

**Comment by sarahscott** [comment]
Part of why this is kind of slow is related to how many rails there are - there are so many `onLayout` calls happening at once. Could be worth thinking about paging at some point?

**Comment by alloy** [comment]
Awesome!

Re performance, first things first, have you tried it on a real device? But even then, I’m sure there are indeed things to optimise, we need to measure and check what is _actually_ happening.

**Comment by sarahscott** [comment]
I haven't tried it on a real device yet but I definitely will today on my iPad. I'm just guessing it has something to do with the number of rails because I isolated a single rail and it was much faster in the sim. Anyway, I'm happy with this PR as it is unless I should look into optimization before merging.

**Comment by sarahscott** [comment]
@alloy is this one mergeable?

**Comment by alloy** [comment]
Yup, let’s do it!

**Pull request merged**